src/python/pants/option/subsystem.py

```diff
@@ -248,6 +250,28 @@ def __eq__(self, other: Any) -> bool:
         return bool(self.options == other.options)
 
 
+class GoalToolMixin(Subsystem):
```

Comment:
This is sort of a misnomer, as `regex-lint` isn't really running a tool, for example. It's legal to write bespoke linting rules that never touch tools.

It also would suggest that `Pytest` should run it, as it's a tool. Or `SetupPy`. But we don't want those to subclass this.

--

Maybe something like `SkippableMixin`..?

Although this IMO speaks to my concern that this is not an appropriate level of abstraction and what we had before was much more explicit and clear.

Reply:
I agree the name can be improved, and happy to hear alternatives. 

Although I don't agree that mypy or pytest shouldn't be skippable. In fact I think it's a bug that they aren't skippable. It makes the CLI less permissive and restricts users to running funnier commands to skip python checking/testing. Likewise if there's plugin checkers or testers (not hard to imagine, with pyright, pyre, and pytype, as well as nose and unittest) we still aren't playing nice.

Reply:
Thoughts on something like `SkippableMixin`?

Reply:
Works for me! I think I've come around to the protocol being nicer. But mypy isn't having it. Let me try it on latest mypy just to be sure 🙂

Reply:
For MyPy, could my `hasattr` suggestion work? I know it's not perfect because you would like MyPy to error when a plugin author left off `skip`, but maybe that is not mission critical?

Reply:
If anything we can use `getattr` without the default and cause a hard error (which the the behavior anyways without a type ignore or just not using mypy).

I feel strongly on the consistency front here.
If that's no bueno, we can check in `__init_subclass__` and error early.